### DISCUSSION TITLE
Should line numbers be right-justified?

### DISCUSSION THREAD

**Discussion by tobydriscoll** [jupyter-book/mystmd#1689] [Q&A]
Maybe this is really a feature request, but when the display of python code shows the wrong indentation, it feels more like a bug.

This is from a `literalinclude` in v1.3.18  with `linenos` set to true.

![Image](https://github.com/user-attachments/assets/9ecccc2a-249b-47b7-a482-4a2dda046a12)

**Comment by rowanc1** [comment]
Agree completely! When developing this the first time, I did try to do this for about an hour.

Will try again soon. :)

**Comment by tobydriscoll** [comment]
It's weirder than I thought. Two separate includes from the same file produce different results:

![image](https://github.com/user-attachments/assets/d5c23504-7b6a-4390-90ec-b6d48106be82)

it doesn't seem to matter which order I do them in. There are no tabs in the source file:

``` python
from scipy.linalg import cholesky
from numpy.linalg import qr
from .FNC02 import forwardsub, backsub

def lsnormal(A,b):
    """
    lsnormal(A,b)

    Solve a linear least squares problem by the normal equations. Returns the
    minimizer of ||b-Ax||.
    """
    N = A.T @ A
    z = A.T @ b
    R = cholesky(N)
    w = forwardsub(R.T, z)                   # solve R'z=c
    x = backsub(R, w)                        # solve Rx=z
    return x

def lsqrfact(A,b):
    """
    lsqrfact(A,b)

    Solve a linear least squares problem by QR factorization. Returns the
    minimizer of ||b-Ax||.
    """
    Q, R = qr(A)
    c = Q.T @ b
    x = backsub(R, c)
    return x
```
